src/components/FlatTreeRenderer.js

```diff
@@ -0,0 +1,436 @@
+const SPRING_CONFIG_GROUP = { clamp: true }
+
+// factor by which second column of table view slides to right
+const TABLE_SECOND_COLUMN_OFFSET = 0
+
+// factor by which first column of table view slides to right
+const TABLE_FIRST_COLUMN_OFFSET = 1.5
+
+const TABLE_FIRST_COLUMN_WIDTH = 8
+
+// factor by which x offset should be increase with each depth
+const DEPTH_OFFSET = 1
+
+/**
+ *
+ */
+const TreeNode = ({
+  styleProps,
+  value,
+  item,
+  heightChangeCallback,
+}) => {
+  const [bind, { height: viewHeight }] = useMeasure()
+  const viewHeightRef = useRef(viewHeight)
+  const wrapperRef = useRef()
+
+  useEffect(() => {
+    heightChangeCallback({ height: viewHeight, key: item.key })
+    viewHeightRef.current = viewHeight
+  }, [viewHeight])
+
+  const isFirstColumn = item.viewInfo.table.column === 1
+
+  /* ### Height Animation ###
+    - viewHeight is 0 at mount and is later populated by the height observer hook.
+    - prevent height animation when second column first item is leaving because
+      y transform animation and overflow hidden with height animation looks weird and janky.
+  */
+
+  const width = isFirstColumn ? `${TABLE_FIRST_COLUMN_WIDTH}rem` : '70%'
+
+  useEffect(() => {
+    // mutating width using ref because changing width using inline style breaks resize observer :(
+    if (wrapperRef.current) wrapperRef.current.style.width = width
+  }, [width])
+
+  useEffect(() => () => heightChangeCallback({ height: null, key: item.key }), [])
+
+  const { x, y, selectionOpacity, rotation, opacity } = styleProps
+
+  return (
+    <animated.div
+      ref={wrapperRef}
+      style={{
+        position: 'absolute',
+        cursor: 'text',
+        height: 0,
+        ...y ? { transform: y.interpolate(_y => `translateY(${_y})`) } : {},
+      }}
+      onClick={() => {
+        store.dispatch({
+          type: 'setCursor',
+          thoughtsRanked: item.path,
+          cursorHistoryClear: true,
+          editing: true,
+        })
+      }}
+    >
+      {/* wrapper div for conistent height observation during re-render because passing bind to animated div causes inconsistency */}
+      <animated.div style={{
+        opacity,
+        ...x ? { transform: x.interpolate(_x => `translateX(${_x})`) } : {},
+      }}>
+        <div {...bind}>
+          <div
+            style={{
+              padding: '0.3rem',
+              display: 'flex',
+            }}
+          >
+            <animated.div
+              style={{
+                height: '0.86rem',
+                width: '0.86rem',
+                marginTop: '0.25rem',
+                borderRadius: '50%',
+                display: 'flex',
+                marginRight: '0.4rem',
+                justifyContent: 'center',
+                alignItems: 'center',
+                ...selectionOpacity ? {
+                  background: selectionOpacity.interpolate(
+                    o => `rgba(255,255,255,${o})`
+                  ) } : {}
```

Comment:
We should be able to set `opacity` directly here since thoughts are now rendered flat. Background `rgba` was used as a hack to prevent style inheritance, otherwise descendants of a thought would incorrectly be affected by the opacity of their ancestors.